azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -300,28 +288,43 @@ public VMImage getImage(final String id) {
 
    @Override
    public Iterable<Location> listLocations() {
+      final Iterable<String> whiteListZoneName = findWhiteListOfRegions();
```

Comment:
Also, better process this in the constructor and assign as a final variable of this class.